sklearn/model_selection/_search.py

```diff
@@ -453,8 +455,18 @@ def score(self, X, y=None):
             raise ValueError("No score function explicitly defined, "
                              "and the estimator doesn't provide one %s"
                              % self.best_estimator_)
-        score = self.scorer_[self.refit] if self.multimetric_ else self.scorer_
-        return score(self.best_estimator_, X, y)
+        if isinstance(self.scorer_, dict):
+            if self.multimetric_:
```

Comment:
@thomasjpfan it seems that this attribute is undocumented. Should it be private instead?

Reply:
I am +0.5 on making it private. (which would require deprecation) Even before this PR, this attribute was defined as:

1. `False` = scoring returns a scalar
2. `True` = scoring returns a dictionary (Which can be a dictionary with one item)

So if `scoring=['accuracy']` that would be considered "multimetric", while `scoring='accuracy'` is not.